From e8f86a026125ff2b2d6bd6eac73d2542852aab84 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Tue, 25 Apr 2017 01:39:51 +0800
Subject: [PATCH 31/41] sunxi: fix the default value of CONS_INDEX on
 non-A23/A33 SUN8I

Only A23/A33 in SUN8I want a default value of CONS_INDEX of 5, for other
chips the default value is 1 like other Allwinner SoCs.

Fix this default value.

The original wrong value has lead to wrong console on H3 Orange Pi
boards.

Fixes: 7095f8641863 ("sunxi: Convert CONS_INDEX to Kconfig")

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
Signed-off-by: Maxime Ripard <maxime.ripard@free-electrons.com>
---
 drivers/serial/Kconfig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/serial/Kconfig b/drivers/serial/Kconfig
index a753367ee1..58320666b7 100644
--- a/drivers/serial/Kconfig
+++ b/drivers/serial/Kconfig
@@ -48,11 +48,11 @@ config CONS_INDEX
 	int "UART used for console"
 	depends on ARCH_SUNXI
 	default 2 if MACH_SUN5I
-	default 5 if MACH_SUN8I
+	default 5 if MACH_SUN8I_A23 || MACH_SUN8I_A33
 	default 1
 	help
 	  Configures the console index.
-	  For Allwinner SoC., default values are 2 for SUN5I and 5 for SUN8I.
+	  For Allwinner SoC., default values are 2 for SUN5I and 5 for A23/A33.
 	  Otherwise, the index equals 1.
 
 config DM_SERIAL
-- 
2.12.2

